pallets/messages/src/lib.rs

```diff
@@ -236,6 +291,17 @@ impl<T: Config> Pallet<T> {
 		})
 	}
 
+	/// Converts a Payload to a message string
+	pub fn payload_to_message(payload: Payload) -> BoundedVec<u8, T::MaxMessagePayloadSizeBytes> {
+		match payload {
+			Payload::Onchain(payload_vec) => payload_vec.try_into().unwrap(),
+			// TODO: What is the right way to format an offchain payload? May need to
+			// table until #190 is picked up.
```

Comment:
This is just internal to the chain right? So I don't think it matters much outside of performance. Perhaps a SCALE encoded struct?